File: yse/device/deviceSetup.cpp
/*
  ==============================================================================

    deviceImplementation.cpp
    Created: 10 Apr 2014 6:05:33pm
    Author:  yvan

  ==============================================================================
*/

#include "deviceSetup.hpp"
#include "../internalHeaders.h"

YSE::DEVICE::setupObject::setupObject()
: in(nullptr),
out(nullptr),
sampleRate(0),
bufferSize(0)
{}

YSE::DEVICE::setupObject & YSE::DEVICE::setupObject::setInput(const interfaceObject & in) {
  this->in = &in;
  return *this;
}

YSE::DEVICE::setupObject & YSE::DEVICE::setupObject::setOutput(const interfaceObject & out) {
  this->out = &out;
  return *this;
}

YSE::DEVICE::setupObject & YSE::DEVICE::setupObject::setSampleRate(double value) {
  sampleRate = value;
  return *this;
}

YSE::DEVICE::setupObject & YSE::DEVICE::setupObject::setBufferSize(int value) {
  bufferSize = value;
  return *this;
}

int YSE::DEVICE::setupObject::getOutputChannels() const {
  if (out == nullptr) return 0;
  return out->getOutputChannelNames().size();
}

File: PCLYSE/DeviceSetup.cs
using System;
using System.Collections.Generic;
using System.Text;
using IYse;

namespace YSE
{
	public class DeviceSetup : Yse.deviceSetup, IYse.IDeviceSetup
	{
		public int GetOutputChannels()
		{
			return getOutputChannels();
		}

		public void SetBufferSize(int value)
		{
			setBufferSize(value);
		}

		public void SetInput(IDevice device)
		{
			setInput((device as Device).device);
		}

		public void SetOutput(IDevice device)
		{
			setOutput((device as Device).device);
		}

		public void SetSampleRate(double value)
		{
			setSampleRate(value);
		}
	}
}
